.github/workflows/build-hpc.yml

```diff
@@ -0,0 +1,108 @@
+name: build-hpc
+
+# Controls when the action will run
+on:
+
+  # Trigger the workflow on all pull requests
+  pull_request: ~
+
+  # Allow workflow to be dispatched on demand
+  workflow_dispatch: ~
+
+jobs:
+  ci-hpc:
+    name: ci-hpc
+
+    strategy:
+      fail-fast: false    # false: try to complete all jobs
+
+      matrix:
+        prec: [dp, sp]
```

Comment:
By default FIELD_API will build both spdp libraries. Only a few files that are precision dependent are compiled twice, the rest are compiled only once. And all the tests are in fact run in DP, expect for one mixed precision test. So we don't need to extend the matrix to cover DP/SP builds. Sorry for missing this earlier.